### DISCUSSION TITLE
12Pピンソケットに何かが詰まっていて12Pピンヘッダが刺さらない

### DISCUSSION THREAD

**Discussion by erm1116** [Yowkees/keyball#284] [Q&A] [answered]
Keyball 61の組み立てを行っています。
ピンソケットにピンヘッダの差し込みを行っているところ、なにかが詰まっており差し込みできない状態です。
こちら解決策などありますでしょか？？

![IMG_5302](https://github.com/Yowkees/keyball/assets/38889285/8238115e-3a87-4dc9-84e8-fd8441995d8f)

**Comment by erm1116** [comment]
なんとか刺さりはし、ピンヘッダの長さがまちまちにはなってしまったのですが、↓のコメントからニッパーで整えれば問題ないとのことでそのまま進めてみました。
https://github.com/Yowkees/keyball/discussions/269 

結果として画像のキーが反応しなさそうなのですが、こちらピンソケット・ピンスイッチ間が原因の可能性などありますでしょうか？
![CleanShot 2023-07-23 at 20 44 30@2x](https://github.com/Yowkees/keyball/assets/38889285/66b7c08b-6ca0-4e6b-bba1-dfc6fe7d83cc)

**Reply by erm1116** [reply]
裏側から見て右側下3つめは、ちょうどROW3部分な気がするのでこれが原因な気がしてきました、、
https://github.com/Yowkees/keyball/blob/main/keyball61/design_data/Circuit%20of%20Keyball61%20Rev1%20Ball_side.pdf

この場合、ピンソケット・ピンヘッダ・ProMicroのつけ直しかなと思うのですが、ピンソケット・ピンヘッダの単体販売は行ってますでしょうか？？

**Comment by Yowkees** [comment, marked as answer]
ご連絡ありがとうございます。
ピンソケット内の異物は、半田付け時に半田が流れ込んでしまったものと思われます。

一列動いていないキーはご指摘の通りROW3の線がProMicroにうまくつながっていないためと思われます。
左手の１キーだけの不良は該当のキーソケットとその付近のダイオードの半田付け不良が濃厚ですので見直してみてください。

ピンソケットは、かろうじて接触している状態になっていると思われ、長期使用で支障が出てきそうですので、新しいものに交換するのが確実です。
Keyballシリーズ購入者向けにピンソケット・ピンヘッダの単体販売可能ですので、一度、下記白銀ラボウェブショップのお問い合わせよりご連絡いただければ幸いです。
https://shirogane-lab.net/inquiry

以上、お手数ですがご対応お願いいたします。

**Reply by erm1116** [reply]
返信ありがとうございます！！
左キーの方は無事に動くようになりました。ありがとうございます！

> ピンソケットは、かろうじて接触している状態になっていると思われ、長期使用で支障が出てきそうですので、新しいものに交換するのが確実です。
Keyballシリーズ購入者向けにピンソケット・ピンヘッダの単体販売可能ですので、一度、下記白銀ラボウェブショップのお問い合わせよりご連絡いただければ幸いです。
[shirogane-lab.net/inquiry](https://shirogane-lab.net/inquiry)

こちら問い合わせの方から連絡させていただきました。よろしくお願いします。
